image_processor.py

```diff
@@ -0,0 +1,198 @@
+
+    def __repr__(self) -> str:
+        return "[Object: x={}; y={}; size={}; distance={}; exists={}]".format(self.x, self.y, self.size, self.distance, self.exists)
+
+
+# results object of image processing. contains coordinates of objects and frame data used for these results
+class ProcessedResults():
+
+    def __init__(self, 
+                balls=[], 
+                basket_b = Object(exists = False), 
+                basket_m = Object(exists = False), 
+                color_frame = [],
+                depth_frame = [],
+                fragmented = [],
+                debug_frame = []) -> None:
+
+
+        self.balls = balls
+        self.basket_b = basket_b
+        self.basket_m = basket_m
+        self.color_frame = color_frame
+        self.depth_frame = depth_frame
+        self.fragmented = fragmented
+
+        # can be used to illustrate things in a separate frame buffer
+        self.debug_frame = debug_frame
+
+
+#Main processor class. processes segmented information
+class ImageProcessor():
+    def __init__(self, camera, color_config = "colors/colors.pkl", debug = False):
+        self.camera = camera
+        self.kernel = np.ones((3, 3), np.uint8)
+
+        self.color_config = color_config
+        with open(self.color_config, 'rb') as conf:
+            self.colors_lookup = pickle.load(conf)
+            self.set_segmentation_table(self.colors_lookup)
+
+        self.fragmented	= np.zeros((self.camera.rgb_height, self.camera.rgb_width), dtype=np.uint8)
+
+        self.t_balls = np.zeros((self.camera.rgb_height, self.camera.rgb_width), dtype=np.uint8)
+        self.t_basket_b = np.zeros((self.camera.rgb_height, self.camera.rgb_width), dtype=np.uint8)
+        self.t_basket_m = np.zeros((self.camera.rgb_height, self.camera.rgb_width), dtype=np.uint8)
+
+        self.debug = debug
+        self.debug_frame = np.zeros((self.camera.rgb_height, self.camera.rgb_width), dtype=np.uint8)
+
+    def set_segmentation_table(self, table):
+        segment.set_table(table)
+
+    def start(self):
+        self.camera.open()
+
+    def stop(self):
+        self.camera.close()
+
+
+    def analyze_balls(self, t_balls, fragments) -> list:
+        t_balls = cv2.dilate(t_balls, self.kernel, iterations=2)
+        #t_balls = cv2.erode(t_balls, self.kernel, iterations=1)
+        contours, hierarchy = cv2.findContours(t_balls, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
+
+        balls = []
+        for contour in contours:
+
+            size = cv2.contourArea(contour)
+
+            if size < 20:
+                continue
+
+            x, y, w, h = cv2.boundingRect(contour)
+
+            ys	= np.array(np.arange(y + h, self.camera.rgb_height), dtype=np.uint16)
+            xs	= np.array(np.linspace(x + w/2, self.camera.rgb_width / 2, num=len(ys)), dtype=np.uint16)
+
+            obj_x = int(x + (w/2))
+            obj_y = int(y + (h/2))
+            obj_dst = obj_y
+
+#black detection
+
+            if (obj_x -4) > 1:
```

Comment:
Can be expressed with `obj_x > 5`. Even better would be `x1 = max(obj_x - 4, 0)`. Then you dont have a need for the if else block. Try to refactor the bottom expressions in a similar fashion